test/unit/org/apache/cassandra/cql3/ViewTest.java

```diff
@@ -1468,24 +1476,44 @@ public void testQuotedIdentifiersInWhereClause() throws Throwable
      * See CASSANDRA-16567 for further details.
      */
     @Test
-    @BMRule(name = "Delay materialized view mutations",
-    targetClass = "StorageProxy",
-    targetMethod = "mutateMV",
-    action = "Thread.sleep(4000);")
+    @BMRules(rules = {
+    @BMRule(name = "Block view builder tasks",
+    targetClass = "ViewBuilderTask",
+    targetMethod = "buildKey",
+    action = "com.google.common.util.concurrent.Uninterruptibles.awaitUninterruptibly" +
+             "(org.apache.cassandra.cql3.ViewTest.blockViewBuild);"),
+    @BMRule(name = "Unblock view builder tasks",
+    targetClass = "ColumnFamilyStore",
+    targetMethod = "truncateBlocking",
+    action = "org.apache.cassandra.cql3.ViewTest.blockViewBuild.countDown();")
+    })
     public void testTruncateWhileBuilding() throws Throwable
     {
         createTable("CREATE TABLE %s (k int, c int, v int, PRIMARY KEY(k, c))");
         execute("USE " + keyspace());
         executeNet("USE " + keyspace());
-        updateView("INSERT INTO %s (k, c, v) VALUES (?, ?, ?)", 0, 0, 0);
+        execute("INSERT INTO %s (k, c, v) VALUES (?, ?, ?)", 0, 0, 0);
         createView("mv",
                    "CREATE MATERIALIZED VIEW %s AS SELECT * FROM %%s " +
                    "WHERE k IS NOT NULL AND c IS NOT NULL AND v IS NOT NULL " +
                    "PRIMARY KEY (v, c, k)");
-        Thread.sleep(2000); // wait for the creation of MV build tasks
+
+        // check that the delayed view builder tasks are running and they haven't written anything yet
+        assertThat(runningViewBuilds()).isPositive();
```

Comment:
From the comment it seems you check only active tasks, but it contains also pending ones